### PR TITLE
Fix Status in Automagic

### PR DESCRIPTION
**Pull request by Spandamn**
Added a function called `canSetStatus` which checks whether a Pokemon can be inflicted with a particular status condition. This should prevent activation of Automagic when a Pokemon cannot be statused by the seconadary effect of a move. @TheImmortal

### CODE REVIEW

File: mods/automagic/scripts.js [hunk 1]
```diff
@@ -254,4 +274,4 @@ exports.BattleScripts = {
 		}
 		return damage;
 	},
-};
+};
```
Comment (Slayer95) [inline mods/automagic/scripts.js]: Please add back the EOF new-line.

Comment (Spandamn) [inline mods/automagic/scripts.js]: oh sorry

Comment (Spandamn) [inline mods/automagic/scripts.js]: Did not know that was intentional

**Comment by Spandamn** [comment]
wait wait dont merge yet

**Comment by Spandamn** [comment]
Shouldve figured this out earlier. mb

**Pull request merged**